aws4.js

```diff
@@ -57,7 +57,7 @@ function RequestSigner(request, credentials) {
 }
 
 RequestSigner.prototype.matchHost = function(host) {
-  var match = (host || '').match(/([^\.]+)\.(?:([^\.]*)\.)?amazonaws\.com$/)
+  var match = (host || '').match(/([^\.]+)\.(?:([^\.]*)\.)?amazonaws\.com(\.cn)*$/)
```

Comment:
I believe this should be `(\.cn)?`

Reply:
Oh yeah, I think you are right. (\.cn)* would also match multiple occurrences of .cn